src/test/java/com/amihaiemil/docker/AuthHttpClientTestCase.java

```diff
@@ -57,6 +62,9 @@ public static void setup() throws Exception {
         noOpClient = Mockito.mock(HttpClient.class);
         Mockito.when(noOpClient.execute(Mockito.any(HttpUriRequest.class)))
             .thenReturn(null);
+        dummyAuth = Mockito.mock(Auth.class);
```

Comment:
@bkuzmic Looks like this mocking is making the tests `injectsHeaderIfAbsent` and `leavesExistingHeaderAlone` never fail

Reply:
@paulodamaso I'm not sure I understand what you mean. In the first test case header is set. In the second, basic header is found in request and not changed.

Reply:
@bkuzmic Let's see: dummyAuth always return `X-Registry-Auth` for `headerName`; it never replaces it with the value, because it alwaus returns the same value each executed test. I'd suggest to use a `Fake` object here where we would instantiate it with no headers in one test and with the correct headre in the second one and then test if those values changed or not, WDYT?

Reply:
@paulodamaso Ok, I agree. I refactored it. Please take a look now.